frontend/src/service/clipboardService/Clipboard.factory.ts

```diff
@@ -0,0 +1,12 @@
+import { IClipboardService } from './Clipboard.service';
+import { NativeClipboardService } from './NativeClipboard.service';
+import { SimulatedClipboardService } from './SimulatedClipboard.service';
+
+export class ClipboardFactory {
+  static createBrowserExtensionService(): IClipboardService {
```

Comment:
```suggestion
 public makeClipboardService(): IClipboardService {
```

Reply:
We want to be able to swap out `ClipboardServiceFactory` implementation during testing using dependency injection. So we can't have static function here.